app/forms/embargo_date_params_filter.rb

```diff
@@ -9,7 +9,11 @@ def call(schema, params)
       next unless dfn[:embargo_date]
 
       name = dfn[:name]
-      date_attributes[name] = deserialize(params, name) if dfn[:assign_if].call(params)
+      date_attributes[name] = if params.slice(*release_params).values.include?('immediate') # rubocop:disable Performance/InefficientHashSearch
```

Comment:
This code helps `nil` out previously set date values if the work switches from an embargo to immediate release. Prior to this change, the correct radio button was selected but a date rendered in the value associated with the unselected radio button. (That is, the "immediately" radio button would be clicked, but an embargo date was still seen beneath it.)